src/modules/roc_pipeline/receiver_session.cpp

```diff
@@ -158,7 +159,9 @@ ReceiverSession::ReceiverSession(const ReceiverSessionConfig& session_config,
 
         resampler_reader.reset(new (allocator) audio::ResamplerReader(
                                    *areader, *resampler_, sample_buffer_pool,
-                                   common_config.internal_frame_size),
+                                   common_config.internal_frame_length,
+                                   common_config.output_sample_rate,
+                                   common_config.output_channels),
```

Comment:
ResamplerReader uses frame_length to split its *input* stream into frames. So we should pass to it the sample rate of the input stream, not of the output stream. So we should replace `common_config.output_sample_rate` with `format->sample_rate` here and above when calling `new_resampler()`.

Reply:
done.